pandas/tseries/tools.py

```diff
@@ -283,10 +303,25 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
     NaT
 
     """
-    return _to_datetime(arg, errors=errors, dayfirst=dayfirst,
-                        yearfirst=yearfirst,
-                        utc=utc, box=box, format=format, exact=exact,
-                        unit=unit, infer_datetime_format=infer_datetime_format)
+    # variable to set offset as per origin parameter
+    offset = None
+
+    if origin == 'julian':
+        if unit != 'D':
+            raise ValueError("unit must be 'D' for origin='julian'")
+        arg = arg - tslib.Timestamp(0).to_julian_date()
```

Comment:
I think you can do this simpler, and check for the origin keyword before the actual conversion (the code you now put in `result_without_offset`. Something like 

```
if unit is not None and offset != 'epoch':
    if origin == 'julian':
        .... (check for unit == D and define offset)
    else:
         offset = ...

    arg += offset